posts/2022-10-24-boa-usage.md

```diff
@@ -0,0 +1,443 @@
+`property`, also for both cases, accessor properties (to use `get` and `set`) and property
+descriptors. You can also get a reference to the `Context` with the `ClasBuilder::context()` method,
+in case you want to do anything fancier.
+
+In this case, the constructor will take care of constructing the Rust `Person` data structure with
+the two arguments it receives, and then register the `say_hello()` method:
+
+```rust
+use boa_engine::{
+    builtins::JsArgs,
+    class::{Class, ClassBuilder},
+};
+
+impl Class for Person {
+    const NAME: &'static str = "Person";
+    const LENGTH: usize = 2;
+
+    // This is what is called when we construct a `Person` with the expression `new Person()`.
+    fn constructor(_this: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<Self> {
+        let name = args.get_or_undefined(0).to_string(context)?;
+        let age = args.get_or_undefined(1).to_u32(context)?;
+
+        if !(0..=150).contains(&age) {
+            context.throw_range_error(format!("invalid age `{age}`. Must be between 0 and 150"))?;
+        }
+
+        let age = u8::try_from(age).expect("we already checked that it was in range");
+
+        let person = Person {
+            name: name.to_string(),
+            age,
+        };
+
+        Ok(person)
+    }
+
+    /// Here is where the class is initialized, to be inserted into the global object.
+    fn init(class: &mut ClassBuilder) -> JsResult<()> {
+        class.method("say_hello", 0, Self::say_hello);
+
+        Ok(())
+    }
+}
+```
+
+In order to register the class, you will need to use the
+[`Context::register_global_class()`][context-class] method:
+
+```rust
+context
+    .register_global_class::<Person>()
+    .expect("could not register class");
+```
+
+You can now adapt the JavaScript code:
+
+```javascript
+let person = new Person("John", 28);
+person.say_hello();
+```
+
+If you want to access the global object from Rust, you can use `Context::global_object()`, which
+will return a [`JsObject`][jsobject]. In this object, you can use the
+[`JsObject::get()`][jsobject-get] function to retrieve any property of the global object, such as
+the `MY_PROJECT_VERSION` that you defined earlier, or any intrinsic, such as the `Date` object.
+
+We are now in the process of creating Rust wrappers for all JavaScript intrinsics ([#2098][2098]).
+For example, you can create a `JsArray` from a `JsObject` to make it much easier to manipulate a
+JavaScript array from Rust. In the following example, you'll create a new `reverseAppend()` global
+function that will receive an array, reverse it, and then append the _"My Project"_ string to it.
+It will then get the `MY_PROJECT_VERSION` from the global object, and append it to the array.
+
+```rust
+use boa_engine::{
+    builtins::JsArgs, object::JsArray, property::Attribute, Context, JsResult, JsValue,
+};
+
+/// Says "hello" using the first argument.
```

Comment:
Wrong comment, assume it was copied from before the other 2

Reply:
Fixed :)